File: php/features.php
<?php
declare(strict_types=1);

// N7timerWeather SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class N7timerWeatherFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new N7timerWeatherBaseFeature();
            case "test":
                return new N7timerWeatherTestFeature();
            default:
                return new N7timerWeatherBaseFeature();
        }
    }
}

File: py/features.py
# N7timerWeather SDK feature factory

from feature.base_feature import N7timerWeatherBaseFeature
from feature.test_feature import N7timerWeatherTestFeature


def _make_feature(name):
    features = {
        "base": lambda: N7timerWeatherBaseFeature(),
        "test": lambda: N7timerWeatherTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
